### PR TITLE
Auto generate ecalls FFI rust code using bindgen in enigma-core enclave

### PR DESCRIPTION
**Pull request by elichai**
Because `build-dependencies` and regular dependencies are mixed together it's not possible to import `bindgen` into the enclave's `build.rs` (https://github.com/rust-lang/cargo/issues/2589)
The solution is to install the `bindgen` CLI in the docker and use it manually in the Makefile.

### CODE REVIEW

File: enigma-core/enclave/Makefile [hunk 1]
```diff
@@ -13,4 +19,12 @@ ifeq ($(XARGO_SGX), 1)
...
\ No newline at end of file
+endif
+
+
+.PHONY: bindgen
+bindgen: Enclave_t.h
+	cargo build -p enigma-types # Meant to make sure `enigma-types.h` already exists and can be included.
+	echo $(BINDGEN_RAW_LINES)
+	bindgen Enclave_t.h $(BINDGEN_FLAGS) -- $(BINDGEN_CLANG_FLAGS) > $(BINDGEN_OUTPUT_FILE)
+	rustfmt  $(BINDGEN_OUTPUT_FILE)
```
Comment (moriaab) [inline enigma-core/enclave/Makefile:32]: Why didn't you use rustfmt option of builder in  #165?

File: enigma-core/enclave/Makefile [hunk 2]
```diff
@@ -13,4 +19,12 @@ ifeq ($(XARGO_SGX), 1)
...
 	cp ./target/$(Rust_target_dir)/libenigmacoreenclave.a ../lib/libenclave.a
-endif
\ No newline at end of file
+endif
+
+
+.PHONY: bindgen
+bindgen: Enclave_t.h
+	cargo build -p enigma-types # Meant to make sure `enigma-types.h` already exists and can be included.
+	echo $(BINDGEN_RAW_LINES)
```
Comment (moriaab) [inline enigma-core/enclave/Makefile]: What is the reason for this echo?

**Review by moriaab — APPROVED** [review]
@elichai Please put the explanation of this PR as a comment in the code

File: enigma-core/enclave/Makefile (hunk 2 above)
Comment (elichai) [reply enigma-core/enclave/Makefile]: opssss

File: enigma-core/enclave/Makefile (hunk 1 above)
Comment (elichai) [reply enigma-core/enclave/Makefile:32]: I'll recheck but I'm pretty sure that it doesn't run on the added `raw_lines` which I add them all at the same line, so `rustfmt` will sort them later nicely

**Pull request merged**